.github/workflows/ci.yml

```diff
@@ -3,19 +3,55 @@ name: ci
 on: [push, pull_request]
 
 jobs:
+  build-bin:
+    runs-on: ubuntu-latest
+    outputs:
+      image-digest: ${{ steps.build-and-export.outputs.digest }}
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and export
+        id: build-and-export
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          outputs: type=docker,dest=/tmp/bin-image.tar
+
+      - name: Upload artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: bin-image
+          path: /tmp/bin-image.tar
```

Comment:
I have to agree with @hayd here, while the resulting image can be slimmed down and simplified a bit, I'm not sure the extra build steps are really necessary. I'm also not sure we really need to introduce `buildx` here as just using the new Dockerfile to create a simple scratch image worked well for me locally. What am I missing here? What are the advantages of using `buildx`?

Reply:
> I'm not sure the extra build steps are really necessary

The `bin` image only needs to be built once, and that's why I don't want to add it to the matrix of distros. This can be optimized later though, by leveraging docker buildx bake, we will not need to use a Matrix in GitHub Actions.

`buildx` is required to use `build-push-action`, which does 2 important things in this context: 

1. outputs to a tar file
2. the image digest as an output variable

Reply:
Let me rephrase: Given the extra steps necessary to build the `bin` image, I'm not sure the added complexity is worth the benefits, especially considering we can get most of the benefits by modifying the current builds to use multi-stage to get rid of some of the filesystem artifacts.

> But when converted to bake, none will be needed.

I'm assuming this is reference to you PR on the main repo?

Reply:
> I'm assuming this is reference to you PR on the main repo?

Kinda, the PR to the main repo has a goal which I don't have much expectations to achieve, but as inspiration, yes.

I'm creating a new commit to this PR. One sec :)

Reply:
@wperron  Well actually it turned out to be more complicated than expected. I will create another PR for it later, and leave this one simpler.